azure/services/agentpools/spec.go

```diff
@@ -224,6 +247,13 @@ func (s *AgentPoolSpec) Parameters(ctx context.Context, existing interface{}) (p
 				Tags:                converters.TagsToMap(s.AdditionalTags),
 			},
 		}
+
+		if kubernetesVersion, err := s.GetManagedMachinePoolVersion(existing); err != nil {
+			return nil, err
+		} else {
```

Comment:
Nit: can drop the else clause and add it directly after `if` block as we return if we enter the `if`.